src/src/com/microsoft/aad/adal/ClientAnalytics.java

```diff
@@ -0,0 +1,101 @@
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package com.microsoft.aad.adal;
+
+import java.util.Map;
+
+/**
+ * Interface that provides ability to client application
+ * send telemetry events to server
+ *
+ * In order to send ADAL instrumentation events to any instrumentation service
+ * client application needs to implement IEventListener interface
+ * and set it for ClientAnalytics
+ * (@use ClientAnalytics.getInstance().setEventListener)
+ */
+public class ClientAnalytics {
+
+    private IEventListener mEventListener;
+
+    /**
+     * Empty constructor, use ClientAnalytics.getInstance()
+     */
+    private ClientAnalytics() {
+    }
+
+    /**
+     * @return event logger
+     */
+    public static ClientAnalytics getInstance() {
+        return InstanceHolder.sInstance;
+    }
+
+    /**
+     * Set custom event listener.
+     * @param eventListener client implementation of instrumentation event logger
+     */
+    public void setEventListener(IEventListener eventListener) {
+        mEventListener = eventListener;
+    }
+
+    /**
+     * send event to listener
+     * @param eventName
+     * @param properties
+     */
+    static void logEvent(final String eventName, Map<String, String> properties) {
+        InstanceHolder.sInstance.log(eventName, properties);
+    }
+
+    /**
+     * if there is event listener set send event to listener
+     * @param eventName
+     * @param properties
+     */
+    private void log(final String eventName, Map<String, String> properties) {
+        if (mEventListener != null) {
+            mEventListener.logEvent(eventName, properties);
+        }
+    }
+
+    /**
+     * Event listener interface
+     */
+    public interface IEventListener {
+
+        /**
+         * Sends event info to server
+         * @param eventName specified name of event that appears of server
```

Comment:
hm. Comment is really strange. I'm not sure what this mean `that appears of server`